### ISSUE TITLE
No official way to install ninja is documented

### ISSUE THREAD

**Issue by marc-hb** [actions/runner-images#11235]
### Description

This is a documentation gap; there is no specific category for that so filing as a "bug".

As detailed in https://github.com/actions/runner-images/issues/514#issuecomment-1915351379 and many others, ninja is probably the most popular tool for building C/C++. It's also used outside C/C++ Unsurprisingly, including `ninja` in the default image has been requested  many times over the years:

- #514
- #741
- #742
- #1531
- #4972
- #8343
- #8348

This has been rejected every time. One of the answers given practically every time: "it's easy to install at run time".

Because `ninja` is the top C/C++ build tool, how to install it should be in the official documentation.
https://docs.github.com/en/search?query=ninja

The official documentation should either:
- provide sample install code that works for each supported operating system + the ${{ conditionals }} to make it portable. Or, much better:
- endorse some official, GitHub-approved action that installs ninja on all operating systems so not every user has to perform a duplicate security audit effort. In other words, officially endorse one of these:
- https://github.com/marketplace/actions/get-cmake
- https://github.com/seanmiddleditch/gha-setup-ninja
- https://github.com/marketplace/actions/install-ninja-build-tool
- ... ?

As of now, installing ninja requires searching closed issues for scattered, conflicting and out of date advice. Random tips to install ninja on Windows vary (either  vcpkg that [comes and goes](https://github.com/actions/runner-images/issues/741#issuecomment-1780114459), unreliable choco, other?), mac recommendations are very slow (brew) and Linux samples miss the important and non-obvious "apt-get update" pre-requirement #2924

From https://github.com/actions/runner-images/issues/8348#issuecomment-1729315414

> "The instrument is extremely easy to install in runtime."

I don't think it is "extremely" easy, see above. Especially not on Windows. But if it truly is, then documenting this will be too.

cc: @erik-bershel, @Darleev, @alepauly, @miketimeofeev, @maxim-lobanov, @dibir-magomedsaygitov

### Platforms affected

- [ ] Azure DevOps
- [X] GitHub Actions - Standard Runners
- [ ] GitHub Actions - Larger Runners

### Runner images affected

- [X] Ubuntu 20.04
- [X] Ubuntu 22.04
- [X] Ubuntu 24.04
- [X] macOS 12
- [X] macOS 13
- [X] macOS 13 Arm64
- [X] macOS 14
- [X] macOS 14 Arm64
- [X] macOS 15
- [X] macOS 15 Arm64
- [X] Windows Server 2019
- [X] Windows Server 2022
- [X] Windows Server 2025

### Image version and build link

This is a documentation gap; there was no category for that.

### Is it regression?

No

### Expected behavior

https://docs.github.com/en/search?query=ninja shows something

### Actual behavior

https://docs.github.com/en/search?query=ninja shows nothing

### Repro steps

https://docs.github.com/en/search?query=ninja

**Comment by RaviAkshintala** [comment]
Hi @marc-hb Thank you for bringing this issue to our attention. We are looking into this issue and will update you on this issue after investigating.

**Comment by Toxe** [comment]
Thank you for bringing this up. (Even though this shouldn't be an issue in the first place.)

**Comment by kelvinphilip889** [comment]
It will be rectify soon

**Comment by vidyasagarnimmagaddi** [comment]
Hi @Toxe ,

- The reference manual  for `ninja` is maintained [here](https://ninja-build.org/manual.html), the source code is maintained in this [github](https://github.com/ninja-build/ninja) repo.

- Also, [docs.github.com](http://docs.github.com/) is open source, if you feel a documentation is missing, please raise a PR with your changes on the [repo](https://github.com/github/docs/blob/main/content/index.md), maintainers would be happy to review your contribution.
-  Ninja can be installed at runtime  in our runner.

We are closing this issue as it does not pertain to github runner images.Thanks.

**Comment by sureshe456** [comment]
Hi @marc-hb, I would like to inform you that we are unable to add these tools in the macOS images due to the need to maintain lightweight operating system images and limited sapce.

We request that you use the below code at runtime for your build. Thanks.
<img width="850" alt="Screenshot 2025-01-01 at 11 38 59 AM" src="https://github.com/user-attachments/assets/eccde435-6635-4fb1-a6b2-e1dded99b1c4" />

**Comment by marc-hb** [comment]
> The reference manual for ninja is maintained [here](https://ninja-build.org/manual.html), the source code is maintained in this [github](https://github.com/ninja-build/ninja) repo.

Thanks but that is irrelevant.

> Ninja can be installed at runtime in our runner.

I don't see where is the information added by this comment.

> Also, [docs.github.com](http://docs.github.com/) is open source, if you feel a documentation is missing, please raise a PR with your changes on the [repo](https://github.com/github/docs/blob/main/content/index.md), maintainers would be happy to review your contribution.

Very good point! However we would first need to know what is the recommended/approved way to install ninja on Windows (mac and Linux seem simpler and consensual).

Also, closing an issue before any work has ever started is strange. 

> We are closing this issue as...
> ... it does not pertain to github runner images.Thanks.

Wow, the second line is probably the most extraordinary so far. There's been a lot of disagreement on this topic but nothing like that yet. This issue is 100% about Github hosted runners; there is really zero doubt about that. Even digressions have stuck about Github hosted runners.

https://docs.github.com/en/actions/using-github-hosted-runners/using-github-hosted-runners/about-github-hosted-runners

> I would like to inform you that we are unable to add these tools in the macOS images due to the need to maintain lightweight operating system images and limited sapce.

ninja is low-level and extremely basic, stable and lightweight. For instance, it is minuscule compared to Visual Studio which is huge and has many (interactive) parts useless on a headless runner. This has been measured and discussed many times, most recently here: https://github.com/actions/runner-images/issues/514#issuecomment-2560076573

Anyway, this issue was NOT asking to revisit this wrong decision; there has already been 6 "wontfix" duplicates for that, see list at the top. This new issue was filed about _mitigating this problem with new documentation_.

> We request that you use the below code at runtime for your build.

Thanks, but:
- That information should be maintained at https://docs.github.com/en/actions/using-github-hosted-runners/using-github-hosted-runners/about-github-hosted-runners, not buried in comments of duplicate issues. Exactly why I filed this.
- That's only for mac. Windows is the difficult one.
- Avoid screenshots of code https://docs.zephyrproject.org/latest/develop/getting_started/index.html#use-copy-paste

**Comment by Toxe** [comment]
@marc-hb It feels like no one has even read anything you have written.

**Comment by marc-hb** [comment]
@Toxe that's a bit extreme but yeah: the huge gap between the time and effort I put versus the level of attention I got is loudly screaming "Don't bother filing tickets".

Well, things could be worse: at least this ticket is public. We get some transparency at least. With some other services, the same would have been buried and hidden in some opaque support system.

**Comment by marc-hb** [comment]
Ninja has apparently been added:
- #11391
